src/client/app/actions/types.ts

```diff
@@ -15,4 +15,12 @@ export type AllActions =
   | HomeActionType
   | DirectoryActionType
 
+export interface ReduxAction<T extends string> {
+  type: T
+}
+
+export interface HasPayload<P> {
+  payload: P
+}
```

Comment:
Suggest defining a single type to save on the repeated use of ampersand, since the keys are always used together.

```typescript
export type ReduxAction<T extends string, P> = {
  type: T,
  payload: P
}
```

Reply:
Initially I tried using this manner of implementing the interface but ran into several issues with the compiler complaining about the void parameters.

In addition, I tried implementing optional parameters and default parameters but those also didn’t work properly.

In the end the current implementation was suggested by Alwyn as a means to accommodate types which had no payload

Reply:
So since the comment is about the &, I would suggest having a second type that just intersects the two together, ie,
```js
export type ReduxActionWithPayload<T extends string, P> = ReduxAction<T> & HasPayload<P>
```

Reply:
> Initially I tried using this manner of implementing the interface but ran into several issues with the compiler complaining about the void parameters.

Very strange. I'd wager that was a strange quirk, and you'll actually be able to get it working if you try again.

Reply:
I tested the single type method again and ran into the same problem. It seems that Typescript expects every parameter within the suggested interface above to be required (truthy), thus providing it with a falsy value such as void (which evaluates to undefined) will throw an error:

`Property 'payload' is missing in type '{ type: "SET_INITIAL_STATE"; }' but required in type 'ReduxActionTest<"SET_INITIAL_STATE", void>`

Reply:
I've implemented LoneRifle's solution in the latest commit